mlflow/store/tracking/file_store.py

```diff
@@ -931,3 +970,81 @@ def record_logged_model(self, run_id, mlflow_model):
             self._set_run_tag(run_info, tag)
         except Exception as e:
             raise MlflowException(e, INTERNAL_ERROR)
+
+
+def _like_pattern_to_regex(pattern):
+    """
+    Convert a pattern string for LIKE and ILIKE to the equivalent regular expression.
+    """
+    return "^" + pattern.replace("%", ".*") + "$"
+
+
+def _comparator_to_func(comparator, value):
+    if comparator == "=":
+        return value.__eq__
+    elif comparator == "!=":
+        return value.__ne__
+    elif comparator in ("LIKE", "ILIKE"):
+        flag = re.IGNORECASE if comparator == "ILIKE" else 0
+        regex = re.compile(_like_pattern_to_regex(value), flag)
+        return regex.match
+    raise MlflowException.invalid_parameter_value(f"Invalid comparator for attribute: {comparator}")
+
+
+def filter_by_attribute(key, comparator, value):
+    f = _comparator_to_func(comparator, value)
+    return lambda exp: f(getattr(exp, key))
+
+
+def filter_by_tags(key, comparator, value):
+    f = _comparator_to_func(comparator, value)
+    return lambda exp: any(f(tag[1]) for tag in filter(lambda tag: tag[0] == key, exp.tags.items()))
```

Comment:
Do we need a loop here?
`exp.tags` should be a dict,
and we can do:
```
if key in exp.tags:
  return f(exp.tags[key])
return False
```

Reply:
Good suggestion, fixed